hkmc2/shared/src/main/scala/hkmc2/syntax/Parser.scala

```diff
@@ -256,14 +264,14 @@ abstract class Parser(
           yeetSpaces match
           case (tok @ BRACKETS(Indent | Curly, toks), loc) :: _ if subRule.blkAlt.isEmpty =>
             consume
-            val blk = rec(toks, S(tok.innerLoc), tok.describe).concludeWith(_.blockOf(subRule, allowNewlines)) // FIXME allowNewlines?
+            val blk = rec(toks, S(tok.innerLoc), tok.describe).concludeWith(_.blockOf(subRule, Nil, allowNewlines)) // FIXME allowNewlines?
             if blk.isEmpty then
               err((msg"Expected ${subRule.whatComesAfter} ${subRule.mkAfterStr}; found end of block instead" -> S(loc) :: Nil))
               errExpr
-            blk ::: blockContOf(rule)
+            blk ::: blockContOf(rule) // TODO: apply headAnnotations
```

Comment:
Not a fan of this. We're going to be silently dropping some information, which is always the worst to debug, if/when we eventually look into why this case is not working.

Do you have a test case that reproduces this path?